### PR TITLE
Fix some SensioLabs Insight violations

### PR DESCRIPTION
**Pull request by dunglas**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | n/a
| License       | MIT
| Doc PR        | n/a

### CODE REVIEW

File: src/Identifier/IdentifierConverter.php [hunk 1]
```diff
@@ -45,13 +45,11 @@ public function convert(string $data, string $class): array
...
 
-        if (!$keys) {
-            throw new InvalidIdentifierException(sprintf('Resource "%s" has no identifiers.', $class));
-        }
-
-        if (\count($keys) > 1) {
+        if ($numIdentifiers = \count($keys) > 1) {
             $identifiers = CompositeIdentifierParser::parse($data);
-        } else {
+        } else if (0 === $numIdentifier) {
```
Comment (dunglas) [inline src/Identifier/IdentifierConverter.php]: `elseif`

File: src/Identifier/IdentifierConverter.php [hunk 2]
```diff
@@ -11,22 +11,21 @@
...
- * Identifier normalizer.
+ * Identifier converter that chains identifier denormalizers.
  *
  * @author Antoine Bluchet <soyuka@gmail.com>
  */
-class ChainIdentifierDenormalizer
+final class IdentifierConverter implements IdentifierConverterInterface
 {
-    const HAS_IDENTIFIER_DENORMALIZER = 'has_identifier_denormalizer';
+    const HAS_IDENTIFIER_CONVERTER = 'has_identifier_converter';
```
Comment (dunglas) [inline src/Identifier/IdentifierConverter.php]: Could be moved to the interface right?

File: src/Identifier/IdentifierConverter.php [hunk 3]
```diff
@@ -40,19 +39,17 @@ public function __construct(IdentifiersExtractorInterface $identifiersExtractor,
...
+    public function convert(string $data, string $class): array
     {
         $keys = $this->identifiersExtractor->getIdentifiersFromResourceClass($class);
 
-        if (!$keys) {
-            throw new InvalidIdentifierException(sprintf('Resource "%s" has no identifiers.', $class));
-        }
-
-        if (\count($keys) > 1) {
+        if ($numIdentifiers = \count($keys) > 1) {
```
Comment (dunglas) [inline src/Identifier/IdentifierConverter.php]: `numIdentifiers` is undefined.

File: src/Identifier/IdentifierConverter.php (hunk 2 above)
Comment (soyuka) [reply src/Identifier/IdentifierConverter.php]: I'd prefer not to so that we can remove it later

File: src/Identifier/IdentifierConverter.php [hunk 4]
```diff
@@ -40,18 +37,16 @@ public function __construct(IdentifiersExtractorInterface $identifiersExtractor,
...
```
Comment (dunglas) [inline src/Identifier/IdentifierConverter.php]: missing parenthesis: `($numIdentifiers = \count($keys))`

**Comment by dunglas** [comment]
Thanks @soyuka

**Pull request merged**